### PR TITLE
MSFT hack week improvements

### PR DESCRIPTION
**Pull request by simra**
Some outputs from hackweek:
- export script to dump images and annotations for ML training
- repair a bug in querying annotations without an image id but with a specific collection id
- add prediction docker image to docker-compose. We might decide this is too resource intensive and have a separate compose script for scenarios where we actually need it.  @koriaf let me know your thoughts

### CODE REVIEW

**Comment by simra** [comment]
PS: what is the magic incantation to bump the UI submodule pointer to main/HEAD?  I can do it locally but don't see how to push it to the remote.

**Comment by simra** [comment]
fixed the UI head issue

File: docker-compose.yml [hunk 1]
```diff
@@ -63,3 +63,10 @@ services:
       - "10000:10000"  # so redirect urls for images work
     logging:
       driver: none
+
+  # this image depends on a custom prebuild script so we can't define the build context here
+  predict_api:
```
Comment (koriaf) [inline docker-compose.yml:68]: docker-compose files support composition, so you can start it like "docker-compose up -f docker-compose.yml -f predict.yml" so frontend developers don't wait for all the related stuff to start

**Review by koriaf — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (simra) [reply docker-compose.yml]: thanks, I pushed a commit to split it out to a separate file.

**Pull request merged**